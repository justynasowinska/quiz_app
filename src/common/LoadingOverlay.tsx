import React from 'react';
import { View, ActivityIndicator, StyleSheet } from 'react-native';

import { THEME_COLORS } from '../config/colors';

interface PropsType {
    backgroundColor?: string;
    size?: 'small' | 'large';
}

export const LoadingOverlay = (props: PropsType) => (
    <View
        style={[
            StyleSheet.absoluteFill,
            styles.loadingOverlay,
            { backgroundColor: props.backgroundColor },
        ]}
    >
        <ActivityIndicator
            color={THEME_COLORS.appTextColor}
            size={props.size || 'small'}
        />
    </View>
);

const styles = StyleSheet.create({
    loadingOverlay: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
        padding: 20,
    },
});
