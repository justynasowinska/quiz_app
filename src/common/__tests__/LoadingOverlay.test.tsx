import * as React from 'react';
import { shallow } from 'enzyme';

import { LoadingOverlay } from '../LoadingOverlay';

describe('LoadingOverlay Component', () => {
    it('matches to the snapshot', () => {
        const component = shallow(<LoadingOverlay backgroundColor="#000" />);

        expect(component).toMatchSnapshot();

        component.setProps({ size: 'large' });
        expect(component).toMatchSnapshot();
    });
});
