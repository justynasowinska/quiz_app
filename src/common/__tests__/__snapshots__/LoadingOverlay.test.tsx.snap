// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingOverlay Component matches to the snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      },
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "padding": 20,
      },
      Object {
        "backgroundColor": "#000",
      },
    ]
  }
>
  <ActivityIndicator
    animating={true}
    color="#474D54"
    hidesWhenStopped={true}
    size="small"
  />
</View>
`;
